README.md

```diff
@@ -12,6 +12,14 @@ The library can be loaded from SQLite as such:
 
 ## Building
 
+### Docker
```

Comment:
Can you add a bit more detail here? The main points to cover are a) that we provide a Docker image with the library pre-built (coming soon to GHCR), b) you can build it yourself if you want, c) how to use the Docker container, i.e. running it loads you into an SQLite repl with the extension already loaded.

It would also be useful to have  just an example showing creating a dummy table (use the code from `tests/success-substruct.sql`), populate it with 1 value, and then run the `substruct_match` function. This should go in the 'Usage' section of the README.